crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -558,54 +562,169 @@ where
 
         let estimated_gas = self
             .estimate_gas_at(
-                CallRequest {
+                TransactionRequest {
                     from: Some(from),
                     to: request.to,
                     gas: request.gas,
-                    gas_price: Some(U256::from(gas_price)),
-                    max_fee_per_gas: Some(U256::from(max_fee_per_gas)),
+                    gas_price: Some(gas_price),
+                    max_fee_per_gas: Some(max_fee_per_gas),
                     value: request.value,
                     input: request.input.clone().into(),
                     nonce: request.nonce,
                     chain_id: Some(chain_id),
                     access_list: request.access_list.clone(),
-                    max_priority_fee_per_gas: Some(U256::from(max_fee_per_gas)),
+                    max_priority_fee_per_gas: Some(max_fee_per_gas),
                     transaction_type: None,
                     blob_versioned_hashes: None,
                     max_fee_per_blob_gas: None,
+                    ..Default::default()
                 },
                 BlockId::Number(BlockNumberOrTag::Pending),
                 None,
             )
             .await?;
         let gas_limit = estimated_gas;
 
-        let transaction = match request.into_typed_request() {
+        let TransactionRequest {
+            to,
+            gas_price,
+            max_fee_per_gas,
+            max_priority_fee_per_gas,
+            gas,
```

Comment:
why is this extra code needed?

if we dedupe types, then this:

let transaction = match request.into_typed_request(

should still work

Reply:
we dont have `into_typed_request` in alloy, typed request doesn't exist there so this was to remove `TransactionRequest` in reth which clashed with the one in alloy. but if u notice i left a comment to remove this before the pr is undrafted

Reply:
I did not notice 🙈 my bad
cool then disregard that